matrix_sdk/src/client.rs

```diff
@@ -646,6 +649,94 @@ impl Client {
         self.send(request).await
     }
 
+    /// Search the homeserver's directory of public rooms.
+    ///
+    /// Returns a `get_public_rooms::Response`, an empty response.
+    ///
+    /// # Arguments
+    ///
+    /// * `limit` - The number of `PublicRoomsChunk`s in each response.
+    ///
+    /// * `since` - Pagination token from a previous request.
+    ///
+    /// * `server` - The name of the server, if `None` the requested server is used.
+    ///
+    /// # Examples
+    /// ```no_run
+    /// use matrix_sdk::{Client, RoomSearchBuilder};
+    /// # use matrix_sdk::api::r0::directory::get_public_rooms;
+    /// # use url::Url;
+    /// # let homeserver = Url::parse("http://example.com").unwrap();
+    /// # let limit = Some(10);
+    /// # let since = Some("since token");
+    /// # let server = Some("server name");
+    ///
+    /// let mut cli = Client::new(homeserver).unwrap();
+    /// # use futures::executor::block_on;
+    /// # block_on(async {
+    /// assert!(cli.get_public_rooms(
+    ///     limit,
+    ///     since,
+    ///     server
+    /// ).await.is_ok());
+    /// # });
+    /// ```
+    pub async fn get_public_rooms(
```

Comment:
Let's remove this get from the name, the spec calls this `publicRooms`, sure it's a `GET` method but the one accepting a filter is a `POST`.